src/plugins/rel/rel.js

```diff
@@ -117,36 +133,25 @@
             }
         }
 
-        var step = {
-                x: toNumber( data.x, prev.x ),
-                y: toNumber( data.y, prev.y ),
-                z: toNumber( data.z, prev.z ),
-                relative: {
-                    x: toNumberAdvanced( data.relX, prev.relative.x ),
-                    y: toNumberAdvanced( data.relY, prev.relative.y ),
-                    z: toNumberAdvanced( data.relZ, prev.relative.z )
-                }
-            };
-
-        // Relative position is ignored/zero if absolute is given.
-        // Note that this also has the effect of resetting any inherited relative values.
-        if ( data.x !== undefined ) {
-            step.relative.x = 0;
-        }
-        if ( data.y !== undefined ) {
-            step.relative.y = 0;
-        }
-        if ( data.z !== undefined ) {
-            step.relative.z = 0;
-        }
-
-        // Apply relative position to absolute position, if non-zero
-        // Note that at this point, the relative values contain a number value of pixels.
-        step.x = step.x + step.relative.x;
-        step.y = step.y + step.relative.y;
-        step.z = step.z + step.relative.z;
-
-        return step;
+        return {
+
+          // If data-? is set, discard any data-rel-? values otherwise use data-rel-?
+          // If neither data-? nor data-rel-? are set - use 0 as a default value
+          x: data.x !== undefined ? toNumber( data.x ) :
+              ( data.relX === undefined ? 0 : prev.x + toNumberAdvanced( data.relX ) ),
```

Comment:
The point of using toNumber() is to keep this block of code simple and easy to read. If you need to do these complex comparisons, create a new function that does all of these nicely:

    x: figureOutCoordinate( ... ),